prime/src/data/actions/Utils.ts

```diff
@@ -1,28 +1,41 @@
 import { isSolvent } from '../BalanceSheet';
-import { MAX_UNISWAP_POSITIONS } from '../constants/Values';
 import { MarginAccount } from '../MarginAccount';
-import { AccountState } from './Actions';
+import { AccountState, OperationResult } from './Actions';
 
 export function runWithChecks(
-  operator: (operand: AccountState) => AccountState | null,
+  operator: (operand: AccountState) => OperationResult,
   operand: AccountState | undefined,
   marginAccount: Omit<MarginAccount, 'assets' | 'liabilities'>
-): AccountState | null {
-  if (operand == null) return null;
+): AccountState {
+  if (operand == null) throw Error('operand is null');
   const updatedOperand = operator(operand);
-  if (updatedOperand == null) return null;
+  if (updatedOperand == null) throw Error('updatedOperand is null');
```

Comment:
I wasn't sure when these really happen so I don't know what the appropriate error messages would be.